### PR TITLE
feat: Enabling CAST.AI features | Workload Autoscaler & Egressd

### PR DESCRIPTION
**Pull request by arunsisodiya**
# Description

As part of this PR, I am proposing changes which can help the module enable a few features that CAST.AI provides.
Below are the features which can be enabled after this PR:
* Workload Autoscaler
* Network Cost Monitoring(Egressd)

Ref: 
https://docs.cast.ai/docs/workload-autoscaling-configuration
https://docs.cast.ai/docs/network-cost

## Type of change

Please delete options that are not relevant.

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -252,6 +252,75 @@ resource "helm_release" "castai_cluster_controller" {
   }
 }
 
+#---------------------------------------------------#
+# CAST.AI Workload Autoscaler configuration         #
+#---------------------------------------------------#
+resource "helm_release" "castai_workload_autoscaler" {
```
[the remark this answers is not in the document]
Comment (arunsisodiya) [reply main.tf:258]: @aldor007 , Yes totally makes sense. I have added the optional flags to install the components.
Changes have been pushed.

**Review by varnastadeus — APPROVED** [review]
If you run `terraform-docs markdown table . --output-file README.md` it should regenerate md docs, other than that looks good 👍

**Comment by arunsisodiya** [comment]
@varnastadeus - Agree, updated the README.md as well with the changes.

If we agree to merge this soon and release it, I can test this out in our environment using the terraform module.
What do you think about that?

File: main.tf [hunk 2]
```diff
@@ -252,6 +252,77 @@ resource "helm_release" "castai_cluster_controller" {
...
+  values  = var.egressd_values
+
+  set {
+    name  = "castai.apiURL"
+    value = var.api_url
+  }
+
+  set {
+    name  = "castai.apiKey"
+    value = var.castai_api_token
```
Comment (varnastadeus) [inline main.tf:311]: this should be `castai_eks_cluster.my_castai_cluster.cluster_token`

Comment (arunsisodiya) [reply main.tf]: Thank you for pointing out @varnastadeus.
Respective changes have been pushed.

**Comment by varnastadeus** [comment]
changes should be available in latest v7.6.0 🎉

**Pull request merged**